geniza/corpus/views.py

```diff
@@ -75,20 +77,27 @@ def get_queryset(self):
 
         self.queryset = documents
 
-        # return 50 documents for now; pagination TODO
-        return documents[:50]
+        return documents
 
-    def get_context_data(self):
-        context_data = super().get_context_data()
-        # should eventually be handled by paginator, but
-        # patch in total number of results for display for now
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+
+        paginator = Paginator(self.queryset, per_page=self.paginate_by)
+        page_num = self.request.GET.get("page", 1)
+        current_page = paginator.page(page_num)
+        paged_result = current_page.object_list
+
+        highlights = paged_result.get_highlighting() if paged_result.count() else {}
+        page_num = int(self.request.GET.get("page", 1))
         context_data.update(
             {
-                "total": self.queryset.count(),
-                "page_title": self.page_title,
+                "highlighting": highlights,
                 "page_description": self.page_description,
+                "page_number": page_num,
```

Comment:
I think it's better to get this from the page object, which is already included in context data; should be `Page.number`